From f909deb4f5d6d394697e1fbbbb4b27c6b5f64038 Mon Sep 17 00:00:00 2001
From: shinchiro <shinchiro@users.noreply.github.com>
Date: Fri, 9 Jun 2017 17:06:18 +0800
Subject: [PATCH] add option for targeting cpu architecture

Because gyp wrongly detect cpu architecture. Also add specific flags
---
 gyp/common.gypi         |  6 ++----
 gyp/common_defines.gypi | 12 +++++++++++-
 2 files changed, 13 insertions(+), 5 deletions(-)

diff --git a/gyp/common.gypi b/gyp/common.gypi
index fec47d84..1f12b114 100644
--- a/gyp/common.gypi
+++ b/gyp/common.gypi
@@ -284,8 +284,7 @@
                         'TargetMachine': '1', # x86
                     },
                 },
-                'defines': [ 'ANGLE_X86_CPU' ],
-            }, # x86_Base
+            },
 
             'x64_Base':
             {
@@ -302,8 +301,7 @@
                         'TargetMachine': '17', # x86 - 64
                     },
                 },
-                'defines': [ 'ANGLE_X64_CPU' ],
-            },    # x64_Base
+            },
 
             # Concrete configurations
             'Debug':
diff --git a/gyp/common_defines.gypi b/gyp/common_defines.gypi
index bce49c4c..c0efaf2c 100644
--- a/gyp/common_defines.gypi
+++ b/gyp/common_defines.gypi
@@ -8,7 +8,7 @@
         'component%': 'static_library',
         'use_libpci%': 1,
         'windows_sdk_path%': 'C:/Program Files (x86)/Windows Kits/10',
-
+        'target_cpu%': 'x86',
         'angle_build_winrt%': '0',
 
         # This works like the Ozone GBM platform in Chrome:
@@ -52,6 +52,16 @@
         {
             'defines': [ 'COMPONENT_BUILD' ],
         }],
+        ['target_cpu=="x86"',
+        {
+            'defines': [ 'ANGLE_X86_CPU' ],
+            'cflags_cc': ['-fomit-frame-pointer', '-mfpmath=sse'],
+        }],
+        ['target_cpu=="x64"',
+        {
+            'defines': [ 'ANGLE_X64_CPU' ],
+            'cflags_cc': ['-fno-omit-frame-pointer'],
+        }],
     ],
     'msvs_settings':
     {
-- 
2.12.1

